proposals/NNNN-transferring-parameters-and-results.md

```diff
@@ -0,0 +1,320 @@
+# `transferring` parameter and result types
+
+* Proposal: [SE-NNNN](NNNN-transferring-parameteres-and-results.md)
+* Authors: [Michael Gottesman](https://github.com/gottesmm), [Holly Borla](https://github.com/hborla), [John McCall](https://github.com/rjmccall)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: On `main` gated behind `-enable-experimental-feature TransferringArgsAndResults`
+* Previous Proposal: [SE-0414: Region-based isolation](/proposals/0414-region-based-isolation.md)
+* Review: ([pitch](https://forums.swift.org/...))
+
+
+## Introduction
+
+This proposal extends region isolation to enable an explicit `transferring`
+annotation to denote when a parameter or result value is required to be in a
+disconnected region, allowing the callee or the caller, respectively, to
+transfer a non-`Sendable` parameter or result value over an isolation boundary.
+
+## Motivation
+
+SE-0414 introduced region isolation to enable safely transferring non-`Sendable`
+values over isolation boundaries. In most cases, function argument and result
+values are merged together into the same region for any given call. This means
+that non-`Sendable` parameter values can never be transferred:
+
+```swift
+// Compiled with -swift-version 6
+
+class NonSendable {}
+
+@MainActor func main(ns: NonSendable) {}
+
+func tryTransfer(ns: NonSendable) async {
+  // error: task isolated value of type 'NonSendable' transferred to
+  // main actor-isolated context
+  await main(ns: ns)
+}
+```
+
+However, for actor initializers, parameters are always considered to be
```

Comment:
This is true for all actor methods. I would just tweak it a little:

```
However for actor methods, parameters are always considered to be transferred into the actor's region since the actor method's body could modify the actor's state using the parameter.
```

Just misc wordsmith.